Content.Shared/Toilet/Systems/SharedToiletSystem.cs

```diff
@@ -0,0 +1,111 @@
+using Content.Shared.Buckle.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Verbs;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+using Content.Shared.Storage.Components;
+using Content.Shared.Toilet.Components;
+
+namespace Content.Shared.Toilet.Systems
+{
+    /// <summary>
+    /// Handles sprite changes for both toilet seat up and down as well as for lid open and closed. Handles interactions with hidden stash
+    /// </summary>
+
+    public abstract class SharedToiletSystem : EntitySystem
+    {
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<ToiletComponent, MapInitEvent>(OnMapInit);
+            SubscribeLocalEvent<ToiletComponent, GetVerbsEvent<AlternativeVerb>>(OnToggleSeatVerb);
+            SubscribeLocalEvent<ToiletComponent, ActivateInWorldEvent>(OnActivateInWorld);
+        }
+
+        private void OnMapInit(EntityUid uid, ToiletComponent component, MapInitEvent args)
+        {
+            EnsureComp<SecretStashComponent>(uid);
+
```

Comment:
You should never be adding components like this on mapinit, use yaml.

Reply:
removed the ensure comp onmapinit.